lib/puppet/provider/service/src.rb

```diff
@@ -63,6 +65,31 @@ def disable
     rmitab(@resource[:name])
   end
 
+  # Wait for the service to transition into the specified state before returning.
+  # This is necessary due to the asynchronous nature of AIX services.
+  # desired_state should either be :running or :stopped.
+  def wait(desired_state)
+    Timeout.timeout(60) do
+      loop do
+        status = self.status
+        break if status == desired_state.to_sym
```

Comment:
You're already passing `desired_state` as a symbol. Is the result of `status` always guaranteed to be a symbol?